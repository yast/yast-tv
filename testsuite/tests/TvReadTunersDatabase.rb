# encoding: utf-8

module Yast
  class TvReadTunersDatabaseClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)

      textdomain "tv"
      Yast.import "Tv"

      @READ_nil = { "target" => { "yast2" => nil } }

      @READ_empty = { "target" => { "yast2" => {} } }

      @READ_db = {
        "target" => {
          "yast2" => {
            "bttv" => [
              { "name" => _("No tuner"), "parameters" => { "tuner" => "4" } },
              { "name" => "Alps HSBH1", "parameters" => { "tuner" => "9" } }
            ]
          }
        }
      }

      # cannot read the database file (creates err-file due to Report::Error)
      # Tv::tuners_database=nil;
      #     TEST (``(Tv::ReadCardsDatabase()), [ READ_nil, $[], $[] ], nil);

      # empty database
      Tv.tuners_database = nil
      TEST(lambda { Tv.ReadTunersDatabase }, [@READ_empty, {}, {}], nil)

      # doesn't read the file (card_database is not nil)
      TEST(lambda { Tv.ReadTunersDatabase }, [@READ_db, {}, {}], nil)

      Tv.tuners_database = nil
      TEST(lambda { Tv.ReadTunersDatabase }, [@READ_db, {}, {}], nil)

      nil
    end
  end
end

Yast::TvReadTunersDatabaseClient.new.main
