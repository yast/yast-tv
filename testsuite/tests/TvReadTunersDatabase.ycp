{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    map READ_nil =
    $[ "target":
		$[ "yast2": nil ]
	];

    map READ_empty =
    $[ "target":
		$[ "yast2": $[] ]
	];

    map READ_db =
	$[ "target": $[
			"yast2": $[
			  "bttv" : [
			    $[
			      "name" : _("No tuner"),
			      "parameters" : $[
			        "tuner" : "4"
			      ]
			    ],
			    $[
			      "name" : "Alps HSBH1",
			      "parameters" : $[
			        "tuner" : "9"
			      ]
			    ]
			  ]
			]
		]
	];

	// cannot read the database file (creates err-file due to Report::Error)
	/*
	Tv::tuners_database=nil;
    TEST (``(Tv::ReadCardsDatabase()), [ READ_nil, $[], $[] ], nil);
	*/

	// empty database
	Tv::tuners_database=nil;
	TEST (``(Tv::ReadTunersDatabase()), [ READ_empty, $[], $[] ], nil);

	// doesn't read the file (card_database is not nil)
	TEST (``(Tv::ReadTunersDatabase()), [ READ_db, $[], $[] ], nil);

	Tv::tuners_database=nil;
    TEST (``(Tv::ReadTunersDatabase()), [ READ_db, $[], $[] ], nil);
}
