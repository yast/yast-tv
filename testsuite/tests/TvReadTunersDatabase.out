Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.Instmode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Read	.target.yast2 "tv_tuners.ycp" $[]
Return	true
Return	true
Read	.target.yast2 "tv_tuners.ycp" $["bttv":[$["name":"No tuner", "parameters":$["tuner":"4"]], $["name":"Alps HSBH1", "parameters":$["tuner":"9"]]]]
Return	true
